messenger.go

```diff
@@ -192,11 +194,11 @@ func (m *Messenger) ProfileByID(id int64, profileFields []string) (Profile, erro
 		qr := QueryResponse{}
 		err = json.Unmarshal(content, &qr)
 		if qr.Error != nil {
-			err = fmt.Errorf("facebook error: %s", qr.Error.Message)
+			return p, xerrors.Errorf("facebook error: %w", qr.Error)
```

Comment:
IMO you should not change the way the control flow works here: while it's clever, it is a bit harder to follow and doesn't really add anything over the way it was done before.

Reply:
Changed it back